tests/Unit/DataStructures/Test_Identity.cpp

```diff
@@ -9,27 +9,34 @@
 #include "tests/Unit/TestHelpers.hpp"
 
 namespace {
-template <size_t Dim>
-void test_identity() {
-  const std::array<double, Dim> used_for_size = make_array<Dim>(0.0);
+template <size_t Dim, typename DataType>
+void test_identity(const DataType& used_for_size) {
   const auto identity_matrix{
-      identity<double, Dim>(dereference_wrapper(used_for_size[0]))};
+      identity<Dim, DataType>(dereference_wrapper(used_for_size))};
```

Comment:
since DataType can be inferred, the second parameter here can be left out. Also remove dereference_wrapper (used_for_size is not a reference_wrapper)